server/src/internalClusterTest/java/org/elasticsearch/cluster/service/ClusterServiceIT.java

```diff
@@ -397,40 +397,48 @@ public void onFailure(Exception e) {
                 fail();
             }
         });
-        invoked3.await();
-
-        for (int i = 2; i <= 5; i++) {
-            clusterService.submitUnbatchedStateUpdateTask(Integer.toString(i), new ClusterStateUpdateTask() {
-                @Override
-                public ClusterState execute(ClusterState currentState) {
-                    return currentState;
-                }
+        assertTrue(invoked3.await(10, TimeUnit.SECONDS));
+
+        try {
+            for (int i = 2; i <= 5; i++) {
+                clusterService.submitUnbatchedStateUpdateTask(Integer.toString(i), new ClusterStateUpdateTask() {
+                    @Override
+                    public ClusterState execute(ClusterState currentState) {
+                        return currentState;
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        fail();
+                    }
+                });
+            }
 
-                @Override
-                public void onFailure(Exception e) {
-                    fail();
-                }
-            });
-        }
-        Thread.sleep(100);
+            final var startNanoTime = System.nanoTime();
+            while (TimeUnit.MILLISECONDS.convert(System.nanoTime() - startNanoTime, TimeUnit.NANOSECONDS) <= 0) {
```

Comment:
This looks weird to me. Does this mean that you wait at least for 1ms to have passed? And if not, you then wait for 100ms (which defeats the purpose of the while loop -- except for rare cases where the sleep is interrupted within 1ms)?

Reply:
Ah re-reading the PR description seems to clarify this now. Indeed you want to ensure that it waits at least 1ms. Couldn't we better though try-catch that thread.sleep for InterruptedException?

Reply:
I encountered a test failure where even the existing 100ms sleep didn't result in `System.nanoTime()` returning a newer time.

I don't think we need to catch an `InterruptedException` here, we can just fail the test in that case and let the test runner work out what to do with the interrupt.